core/state_transition.go

```diff
@@ -193,7 +193,7 @@ func (st *StateTransition) buyGas() error {
 	if st.gasFeeCap != nil {
 		balanceCheck = new(big.Int).SetUint64(st.msg.Gas())
 		balanceCheck = balanceCheck.Mul(balanceCheck, st.gasFeeCap)
-		balanceCheck.Add(balanceCheck, st.value)
+		balanceCheck = balanceCheck.Add(balanceCheck, st.value)
```

Comment:
I would fix it the other way around, get rid of `balanceCheck =` for the multiplication.

Reply:
It makes the code clearer for future modifications, otherwise with the "double" assignment, someone might interpret is at a new return instead of modifying the original.

Reply:
That's a good point that it might confuse people into thinking that the assignment is required. I mistakenly thought that the convention used throughout the codebase was to do the double assignment and thought that made it look much clearer in terms of the math operations that are going on, but understand your point, will close this PR.